interpreter/interpreter.go

```diff
@@ -613,6 +613,7 @@ func (itrp *Interpreter) doCall(ctx context.Context, call *semantic.CallExpressi
 	}
 
 	// Call the function
+	ctx = context.WithValue(ctx, "scope", scope)
```

Comment:
This seems wrong to me, but I'm not sure what the proper way would be to access options.

If we do end up using context to pass the current scope, we would need to create a specific context key for it and formalize this as the way to access the surrounding scope.

Another way we can do this is by requiring the now time to be passed to the function. I've used this pattern a few times:

```
truncate = (t, unit) => _truncate(t: t, unit: unit, now: now())
builtin _truncate
```

This way the resolution of the `now()` identifier is handled by the interpreter instead of within Go. This is a possible way to deal with this as I don't think this is something we've encountered previously.

I also know @adrian-thurston was working on a way of injecting the now time into the context for `tableFind`. I don't know if he ever merged that but we may want to consider that as a possible solution too.